report-api/src/test/java/no/sikt/nva/data/report/api/fetch/FetchNviInstitutionReportHandlerPresignerTest.java

```diff
@@ -0,0 +1,156 @@
+    private static final String OPEN_XML = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
+    private static final String EXCEL = "application/vnd.ms-excel";
+    private static final String SOME_YEAR = "2023";
+    private static final AccessRight SOME_ACCESS_RIGHT_THAT_IS_NOT_MANAGE_NVI = AccessRight.SUPPORT;
+    private FetchNviInstitutionReportPresigner handler;
+    private FakeSqsClient queueClient;
+    private S3Presigner mockedS3Presigner;
+
+    @BeforeEach
+    public void setup() {
+        queueClient = new FakeSqsClient();
+        mockedS3Presigner = mock(S3Presigner.class);
+        handler = new FetchNviInstitutionReportPresigner(queueClient, mockedS3Presigner);
+    }
+
+    @Test
+    void shouldReturn401WhenUserDoesNotHaveManageNviAccessRight() throws IOException {
+        var request = new FetchNviInstitutionReportProxyRequest(SOME_YEAR, TEXT_PLAIN);
+        var unAuthorizedRequest = generateHandlerRequest(request, SOME_ACCESS_RIGHT_THAT_IS_NOT_MANAGE_NVI,
+                                                         randomUri());
+        var output = new ByteArrayOutputStream();
+        var context = new FakeContext();
+        handler.handleRequest(unAuthorizedRequest, output, context);
+        var response = fromOutputStream(output, GatewayResponse.class);
+        var actualProblem = objectMapper.readValue(response.getBody(), Problem.class);
+        var expectedProblem = getExpectedProblem(context.getAwsRequestId());
+        assertEquals(expectedProblem, objectMapper.writeValueAsString(actualProblem));
+    }
+
+    @Test
+    void shouldLogRequestingUsersTopLevelOrganization() throws IOException {
+        var logAppender = LogUtils.getTestingAppenderForRootLogger();
+        var institutionId = randomUri();
+        var request = generateHandlerRequest(new FetchNviInstitutionReportProxyRequest(SOME_YEAR, TEXT_PLAIN),
+                                             institutionId);
+        var output = new ByteArrayOutputStream();
+        var context = new FakeContext();
+        handler.handleRequest(request, output, context);
+        assertTrue(logAppender.getMessages().contains("for organization: " + institutionId));
+    }
+
+    @Test
+    void shouldExtractAndLogPathParameterReportingYear() throws IOException {
+        var logAppender = LogUtils.getTestingAppenderForRootLogger();
+        var request = generateHandlerRequest(new FetchNviInstitutionReportProxyRequest(SOME_YEAR, TEXT_PLAIN),
+                                             randomUri());
+        var output = new ByteArrayOutputStream();
+        var context = new FakeContext();
+        handler.handleRequest(request, output, context);
+        assertTrue(logAppender.getMessages().contains("reporting year: " + SOME_YEAR));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {TEXT_CSV, TEXT_PLAIN, OPEN_XML, EXCEL})
+    void shouldReturnPreSignedUrl(String contentType) throws IOException {
```

Comment:
Test for non-supported content types?

Reply:
Do you mean wrong code `302` in commons @brinxmat?

Reply:
Then all is good.